staging/src/k8s.io/apiserver/pkg/util/flowcontrol/match_test.go

```diff
@@ -120,162 +120,236 @@ func TestLiterals(t *testing.T) {
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
 			User: &fcv1a1.UserSubject{"goodu"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:      []string{"goodverb"},
-			APIGroups:  []string{"goodapig"},
-			Resources:  []string{"goodrscs"},
-			Namespaces: []string{"goodns"}}}}, {
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"goodrscs"},
+			Namespaces:    []string{"goodns"},
+			ResourceNames: []string{"eman"},
```

Comment:
Given that we are now distinguishing between matching and mismatching values for ResourceName, I would not object if the value here were `"goodn"` rather than `"eman"` (with corresponding changes in the test cases).

Reply:
fixed